gen/map.go

```diff
@@ -190,6 +190,101 @@ func (m *mapGenerator) Reader(g Generator, spec *compile.MapSpec) (string, error
 	return name, wrapGenerateError(spec.ThriftName(), err)
 }
 
+// Equals generates a function to compare maps of the given type
+//
+// 	func $name(lhs, rhs $mapType) bool {
+// 		...
+// 	}
+//
+// And returns its name.
+func (m *mapGenerator) Equals(g Generator, spec *compile.MapSpec) (string, error) {
+	name := "_" + valueName(spec) + "_Equals"
+
+	err := g.EnsureDeclared(
+		`
+			<$mapType := typeReference .Spec>
+
+			<$lhs := newVar "lhs">
+			<$rhs := newVar "rhs">
+			func <.Name>(<$lhs>, <$rhs> <$mapType>) bool {
+				if len(<$lhs>) != len(<$rhs>) {
+					return false
+				}
+
+				<$lk := newVar "lk">
+				<$lv := newVar "lv">
+				<$rv := newVar "rv">
+				<$ok := newVar "ok">
```

Comment:
nit: You can actually move these into the `<if ..>` block.

That will also remove them once they drop out of scope.